src/ElvContracts.js

```diff
@@ -246,6 +246,132 @@ class ElvContracts {
     return {total: Ethers.BigNumber.from(res._hex).toNumber()};
   }
 
+  /**
+   * Deploy Payment contract (revenue splitter - commerce/Payment.sol)
+   * @param {string} addresses : list of stakeholder addresses
+   * @param {string} shares: list of stakeholder shares, in the order of addresses
+   */
+  async PaymentDeploy({ addresses, shares }){
```

Comment:
Does the contract check for bad share values?

Reply:
These are the checks done in constructor:
```
require(payees.length == shares_.length, "PaymentSplitter: payees and shares length mismatch");
require(payees.length > 0, "PaymentSplitter: no payees");


require(account != address(0), "PaymentSplitter: account is the zero address");
require(shares_ > 0, "PaymentSplitter: shares are 0");
require(_shares[account] == 0, "PaymentSplitter: account already has shares");
```

Reply:
I added validation for addresses and shares (integers)